libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -690,6 +690,13 @@ namespace impl {
                   std::get<additional_block_signatures_extension>(block_exts.lower_bound(additional_block_signatures_extension::extension_id())->second);
             mvo("additional_signatures", additional_signatures);
          }
+         auto qc_extension_count = block_exts.count(quorum_certificate_extension::extension_id());
+         if ( qc_extension_count > 0) {
+            EOS_ASSERT(qc_extension_count == 1, ill_formed_quorum_certificate_extension, "At most one quorum certificate is allowed in block extension, which has ${c}.", ("c", qc_extension_count));
```

Comment:
Should not assert here.

Reply:
We have a requirement that at most one QC is allowed. Where is a better place to check?

Reply:
At time of insertion into the block.

Reply:
The extension definition has:
`static constexpr bool     enforce_unique() { return true; }`
so I assume there cannot be more than one.